samples/ClientSample/HubSample.cs

```diff
@@ -37,30 +37,21 @@ internal static void Register(CommandLineApplication app)
             try
             {
 
-                var cts = new CancellationTokenSource();
-                Console.CancelKeyPress += (sender, a) =>
+                Console.CancelKeyPress += async (sender, a) =>
                 {
                     a.Cancel = true;
                     Console.WriteLine("Stopping loops...");
-                    cts.Cancel();
+                    await connection.DisposeAsync();
                 };
 
                 // Set up handler
                 connection.On<string>("Send", Console.WriteLine);
+                var cts = new CancellationTokenSource();
```

Comment:
Is this being used at all?

Reply:
It's being passed into InvokeAsync here https://github.com/aspnet/SignalR/pull/1080/files/9cc475fc9fbf74d2b8b6778f0e979c52f812e80c#diff-c044366a1200519c9bd2ab30a0243a34R66 but it also needs to be canceled in CancelKeyPress